src/main/java/com/auth0/TransientCookieStore.java

```diff
@@ -0,0 +1,154 @@
+package com.auth0;
+
+import org.apache.commons.lang3.Validate;
+
+import javax.servlet.http.Cookie;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * Allows storage and retrieval/removal of cookies.
+ */
+class TransientCookieStore {
+
+    private static final int MAX_AGE_SECONDS = 600; // 10 minutes
+
+    // Prevent instantiation
+    private TransientCookieStore() {}
+
+
+    /**
+     * Stores a state value as a cookie on the response.
+     *
+     * @param response the response object to set the cookie on
+     * @param state the value for the state cookie. If null, no cookie will be set.
+     * @param sameSite the value for the SameSite attribute on the cookie
+     * @param legacySameSiteCookie whether to set a fallback cookie or not
+     */
+    static void storeState(HttpServletResponse response, String state, SameSite sameSite, boolean legacySameSiteCookie) {
+        store(response, StorageUtils.STATE_KEY, state, sameSite, legacySameSiteCookie);
+    }
+
+    /**
+     * Stores a nonce value as a cookie on the response.
+     *
+     * @param response the response object to set the cookie on
+     * @param nonce the value for the nonce cookie. If null, no cookie will be set.
+     * @param sameSite the value for the SameSite attribute on the cookie
+     * @param legacySameSiteCookie whether to set a fallback cookie or not
+     */
+    static void storeNonce(HttpServletResponse response, String nonce, SameSite sameSite, boolean legacySameSiteCookie) {
+        store(response, StorageUtils.NONCE_KEY, nonce, sameSite, legacySameSiteCookie);
+    }
+
+    /**
+     * Gets the value associated with the state cookie and removes it.
+     * @param request the request object
+     * @param response the response object
+     * @param legacySameSiteCookie whether to use a fallback cookie or not
+     * @return the value of the state cookie, if it exists
+     */
+    static String getState(HttpServletRequest request, HttpServletResponse response, boolean legacySameSiteCookie) {
+        return getOnce(StorageUtils.STATE_KEY, request, response, legacySameSiteCookie);
+    }
+
+    /**
+     * Gets the value associated with the nonce cookie and removes it.
+     * @param request the request object
+     * @param response the response object
+     * @param legacySameSiteCookie whether to use a fallback cookie or not
+     * @return the value of the nonce cookie, if it exists
+     */
+    static String getNonce(HttpServletRequest request, HttpServletResponse response, boolean legacySameSiteCookie) {
+        return getOnce(StorageUtils.NONCE_KEY, request, response, legacySameSiteCookie);
+    }
+
+    private static void store(HttpServletResponse response, String key, String value, SameSite sameSite, boolean legacySameSiteCookie) {
+        Validate.notNull(response, "response must not be null");
```

Comment:
On other methods you do `Validate.notNull(response)` without passing the message

Reply:
I updated other validations to pass a message as well.